script.js

```diff
@@ -1,4 +1,138 @@
+var rollDiceForPlayer = function () {
+  var counter = 0;
+  while (counter < 2) {
+    currentPlayerRolls.push(rollDice());
+    counter = counter + 1;
+  }
+  return (
+    "Hi Battle Player <br><br> You rolled <br> Dice 1: " +
+    currentPlayerRolls[0] +
+    " & Dice 2: " +
+    currentPlayerRolls[1] +
+    "<br><br> Please input either '1' or '2' to choose the dice for the first digit value."
+  );
+};
+
+//Function to get player's score
+var getPlayerScore = function (playerInput) {
+  var playerScore;
+  if (playerInput != 1 && playerInput != 2) {
+    return (
+      "Oops! There is an error. Please input 1 or 2 only. Your dice rolls were: <br> Dice 1: " +
+      currentPlayerRolls[0] +
+      " Dice 2: " +
+      currentPlayerRolls[1]
+    );
+  }
+  if (playerInput == 1) {
+    playerScore = Number(
+      String(currentPlayerRolls[0]) + String(currentPlayerRolls[1])
+    );
+  }
+  if (playerInput == 2) {
+    playerScore = Number(
+      String(currentPlayerRolls[1]) + String(currentPlayerRolls[0])
+    );
+  }
+  allPlayersScore.push(playerScore);
+  currentPlayerRolls = [];
+
+  return (
+    "Player" +
+    " " +
+    currentPlayer +
+    " " +
+    "Your chosen value is: " +
+    playerScore
+  );
+};
+
+// Function to compare player 1 and player 2 scores
+var comparePlayersScore = function () {
+  var compareMessage =
+    "Player 1 score: " +
+    allPlayersScore[0] +
+    "<br> Player 2 score: " +
+    allPlayersScore[1];
+  if (allPlayersScore[0] > allPlayersScore[1]) {
+    compareMessage = compareMessage + "<br><br> Player 1 wins!";
+  }
+  if (allPlayersScore[1] > allPlayersScore[0]) {
+    compareMessage = compareMessage + "<br><br> Player 2 wins!";
+  }
+  if (allPlayersScore[0] == allPlayersScore[1]) {
+    compareMessage =
+      compareMessage + "<br><br> It is a tie. You guys deserve a prize!";
+  }
+  return compareMessage;
+};
+
+// Function to reset game
+var resetGame = function () {
+  currentPlayer = 1;
+  gameState = GAME_STATE_DICE_ROLL;
+  allPlayersScore = [];
+};
+
+// Main function
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var myOutputValue = " ";
+  if (gameState == GAME_STATE_DICE_ROLL) {
+    myOutputValue = rollDiceForPlayer();
+    gameState = GAME_STATE_CHOOSE_DICE_ORDER;
+    return myOutputValue;
+  }
+
+  if (gameState == GAME_STATE_CHOOSE_DICE_ORDER) {
+    myOutputValue = getPlayerScore(input);
+    if (currentPlayer == 1) {
+      currentPlayer = 2;
+      gameState = GAME_STATE_DICE_ROLL;
+      return myOutputValue + "<br><br> It is Player 2's turn now! ";
+    }
+
+    if (currentPlayer == 2) {
+      gameState = GAME_STATE_COMPARE_SCORES;
+      return (
+        myOutputValue + "<br><br> Press submit to calculate the scores :) "
+      );
+    }
+  }
+
+  if (gameState == GAME_STATE_COMPARE_SCORES) {
+    myOutputValue = comparePlayersScore();
+    resetGame(); // to reset game
+    return myOutputValue;
+  }
 };
```

Comment:
All in all, a really good attempt! Good use of functions to make your main function clean and neat. The next step is to try to think of how to make functions reusable. The key to that is making functions in a way that does one simple task. Knowing exactly when and how to break functions into smaller pieces takes experience and practice so don't worry about it! Keep up the good work 👍